src/relop.cpp

```
@@ -264,5 +264,241 @@ ScalarFunctionSet base_r_gte() {
+			result.SetVectorType(VectorType::CONSTANT_VECTOR);
+			auto result_data = ConstantVector::GetData<bool>(result);
+			*result_data = is_in_y(*ConstantVector::GetData<LHS_TYPE>(x));
+
+			break;
+		}
+
+		default : {
+			UnifiedVectorFormat vdata;
+			x.ToUnifiedFormat(count, vdata);
+			result.SetVectorType(VectorType::FLAT_VECTOR);
+			in_loop(
+				count,
+				FlatVector::GetData<LHS_TYPE>(x),
+				FlatVector::GetData<bool>(result),
+				vdata.validity
+			);
+
+			break;
+		}
+	}
+
+}
+
+#define IN_VARIANT(__LHS__, __RHS__) ScalarFunction(                                      \
+	/* arguments   = */ {LogicalType::__LHS__, LogicalType::LIST(LogicalType::__RHS__)},  \
+	/* return_type = */ LogicalType::BOOLEAN,                                \
+	/* function    = */ InExecute<                                           \
+		typename physical<LogicalType::__LHS__>::type, \
+	    typename physical<LogicalType::__RHS__>::type  \
+	>)
+
+} // namespace
+
+ScalarFunctionSet base_r_in() {
+	ScalarFunctionSet set("r_base::%in%");
+
+	set.AddFunction(IN_VARIANT(DOUBLE, DOUBLE));
+	set.AddFunction(IN_VARIANT(BOOLEAN, BOOLEAN));
+	set.AddFunction(IN_VARIANT(BOOLEAN, INTEGER));
+	set.AddFunction(IN_VARIANT(INTEGER, BOOLEAN));
+	set.AddFunction(IN_VARIANT(INTEGER, INTEGER));
+
+	set.AddFunction(IN_VARIANT(DOUBLE, INTEGER));
+	set.AddFunction(IN_VARIANT(INTEGER, DOUBLE));
+	set.AddFunction(IN_VARIANT(DOUBLE, BOOLEAN));
+	set.AddFunction(IN_VARIANT(BOOLEAN, DOUBLE));
+
+	set.AddFunction(IN_VARIANT(VARCHAR, VARCHAR));
+	// set.AddFunction(IN_VARIANT(VARCHAR, INTEGER));
+	// set.AddFunction(IN_VARIANT(INTEGER, VARCHAR));
+	// set.AddFunction(IN_VARIANT(VARCHAR, DOUBLE));
+	// set.AddFunction(IN_VARIANT(DOUBLE, VARCHAR));
+	set.AddFunction(IN_VARIANT(VARCHAR, BOOLEAN));
+	set.AddFunction(IN_VARIANT(BOOLEAN, VARCHAR));
+
+	set.AddFunction(IN_VARIANT(TIMESTAMP, TIMESTAMP));
+	set.AddFunction(IN_VARIANT(TIMESTAMP, VARCHAR));
+	set.AddFunction(IN_VARIANT(VARCHAR, TIMESTAMP));
```

These two variants should throw a binding error, this means that duckplyr will forward this to R.